### PR TITLE
create a foundation for interface target compile on android

### PR DESCRIPTION
**Pull request by birarda**
- CMake changes to setup android build of interface target
- tweaks in various libraries for standalone compile as shared libraries
- android toolchain changes for full C++11 support

### CODE REVIEW

**Comment by birarda** [comment]
#### Test Plan

Because this touches CMake files that the desktop targets use to compile, please do a smoke test.

**Comment by hifi-gustavo** [comment]
Build console output for commit: ae87f966c8e6938a545477379de4ffe7cf538e99 available at: https://gist.github.com/e260dee660fe5cff1014a7c1be35eca6<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8559-ae87f96.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8559-ae87f96.dmg

**Comment by birarda** [comment]
Forgot that VS2013 doesn't have `constexpr` support. Need to fix windows build.

File: libraries/recording/src/recording/Frame.h [hunk 1]
```diff
@@ -22,7 +27,13 @@ namespace recording {
 struct FrameHeader {
     using Time = uint32_t;
 
+// until we use a version of visual studio that has constexpr support, we can't use numeric_limits at compile time
```
Comment (ZappoMan) [inline libraries/recording/src/recording/Frame.h]: boo!

**Comment by hifi-gustavo** [comment]
Build console output for commit: 545181a465c432048712009d79f5c08fd32cd06f available at: https://gist.github.com/e260dee660fe5cff1014a7c1be35eca6<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8559-545181a.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8559-545181a.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: f48da0252ad492209e77b7e43a02a8f92ecaa4ed available at: https://gist.github.com/e260dee660fe5cff1014a7c1be35eca6<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8559-f48da02.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8559-f48da02.dmg

**Comment by kevinmthomasse** [comment]
testing

**Comment by kevinmthomasse** [comment]
## QA Test
**QA Tester**: KevinMThomas
**localhost 'createTestBlocks.js' Result**: **localhost 'createTestBlocks.js'** SUCCESS
**localhost ATP Upload + Edit Result**: **localhost ATP** SUCCESS
**Echo Server Audio**: **Echo Server Audio** SUCCESS
**localhost 'harmonicOscillator.js' Result**: **localhost 'harmonicOscillator.js'** SUCCESS
**localhost 'reverbTest.js' Result**: **localhost 'reverbTest.js'** SUCCESS
**localhost 'playaPerformanceTest.js' Result**: **localhost 'playaPerformanceTest.js'** SUCCESS
**dev-playa 'playaPerformanceTest.js' Result**: **dev-playa 'playaPerformanceTest.js'** SUCCESS
**dev-demo 'playaPerformanceTest.js' Result**: **dev-demo 'playaPerformanceTest.js'** SUCCESS
**QA Pass**: **QA Pass** SUCCESS
**Additional Test Results + Observations**: Successful test.


**Domain** |localhost |dev-playa |dev-demo
---|---|---|---
**Render Rate**: |59 |60 |60
**Present Rate**: |60 |60 |58
**Present New Rate**: |0 |0 |0
**CPU Buffer Count**: |886 |3703 |1934
**GPU Buffer Count**: |314 |851 |434
**CPU Memory Usage**: |24125440 |70893568 |38752256
**GPU Memory Usage**: |16130048 |26619904 |23568384
**CPU Texture Count**: |53 |276 |232
**GPU Texture Count**: |1161 |1393 |1443
**GPU Transfer Count**: |0 |0 |0
**CPU Texture Memory Usage**: |20501336 |653861720 |251154520
**GPU Texture Memory Usage**: |88264772 |1131074482 |519578452
**GPU Virtual Memory Usage**: |214036380|1256846090|645350060
**Triangle Count**: |108848 |895211 |181319
**Triangle Rate**: |6402823 |55950687 |11332437
**API Drawcall Count**: |27 |301 |87
**Drawcall Count**: |27 |344 |87
**Drawcall Rate**: |1588 |21500 |5437
**State Changes Texture Count**:|48 |320 |194
**Pipeline Count**: |33 |41 |34
**Item Opaques**: |6 |463 |12
**Item Translucents**: |11 |26 |69
**Item Lights**: |0 |77 |1
**Timing Opaques**: |55 |1450 |105
**Timing Translucents**: |25 |50 |209
**Timing Lighting**: |122 |84 |175
**Render Frame**: |840 |2597 |990

**Pull request merged**